elementary/clients/dbt/slim_dbt_runner.py

```diff
@@ -126,23 +129,41 @@ def _load_config(self):
         self.config = RuntimeConfig.from_args(self.args)
 
     def _load_adapter(self):
+        if not self.config:
+            raise Exception("Config not loaded")
+
         register_adapter(self.config)
         self.adapter_name = self.config.credentials.type
-        self.adapter = get_adapter_class_by_name(self.adapter_name)(self.config)
-        self.adapter.connections.set_connection_name()
-        self.config.adapter = self.adapter
+        self.adapter = cast(
+            BaseAdapter, get_adapter_class_by_name(self.adapter_name)(self.config)
+        )
```

Comment:
Casts are needed here because dbt use `Protocol` classes as the return value types, and not the actual base classes
And these classes don't actually contain some of the methods we use here (which are internal so probably not a part of the official adapter protocol)